lib/test.js

```diff
@@ -66,34 +67,39 @@ Test.prototype.skip = function () {
 	this.skipTest = true;
 };
 
-Test.prototype.run = function (cb) {
-	this.cb = cb;
+Test.prototype.run = function () {
+	this.promise = {};
 
-	if (!this.fn || this.skipTest) {
-		this.exit();
-	}
-
-	this._timeStart = Date.now();
+	return new Promise(function (resolve, reject) {
+		this.promise.resolve = resolve;
+		this.promise.reject = reject;
```

Comment:
Ok. Can you at least add a `TODO` in there so it's not forgotten that it needs improving?